numba/_lapack.c

```diff
@@ -486,12 +507,52 @@ numba_xxpotrf(char kind, char uplo, Py_ssize_t n, void *a, Py_ssize_t lda)
     _lda = (F_INT) lda;
 
     (*(xxpotrf_t) raw_func)(&uplo, &_n, a, &_lda, &info);
-    CATCH_LAPACK_INVALID_ARG(info);
+    CATCH_LAPACK_INVALID_ARG("xxpotrf", info);
     return info;
 }
 
 
 /*
+ * kind_size()
+ * gets the data size appropriate for a specified kind.
+ *
+ * Input:
+ * kind - the kind, one of:
+ *         (s, d, c, z) = (float, double, complex, double complex).
+ *
+ * data_size - modified in place, on return contains the appropriate data size.
+ *
+ * Returns 0 on success, -1 else.
```

Comment:
I'm talking about the code calling kind_size(). It doesn't "propagate" the error, it returns -1 explicitly.
Anyway, this is more of a detail.